actions/show-heartbeat/index.js

```diff
@@ -3,6 +3,92 @@ import {Action, registerAction, weightedChoose} from '../utils';
 const VERSION = 52; // Increase when changed.
 const LAST_SHOWN_DELAY = 1000 * 60 * 60 * 24 * 7; // 7 days
 
+
+export class HeartbeatFlow {
+    constructor(action) {
+        this.action = action;
+
+        let flashPlugin = navigator.plugins['Shockwave Flash'];
+        let plugins = {};
+        for (let plugin of Array.from(navigator.plugins)) {
+            plugins[plugin.name] = plugin.version;
+        }
+
+        let {normandy, recipe, survey, client, location} = action;
+        this.data = {
+            // Required fields
+            response_version: 2,
+            experiment_version: '-',
+            person_id: 'NA',
+            survey_id: recipe.arguments.surveyId,
+            flow_id: normandy.uuid(),
+            question_id: survey.message,
```

Comment:
Nope. The question ID used by the current self-repair is the question text as well: https://github.com/mozilla/self-repair-server/blob/master/src/recipes/heartbeat-by-user-first-impression/index.js#L148